src/solver/hydro/management/management.h

```diff
@@ -100,11 +102,12 @@ class HydroManagement final
     // \return The total inflow for the whole year
     double prepareMonthlyTargetGenerations(Data::Area& area, PerArea& data);
 
-    void prepareDailyOptimalGenerations(Solver::Variable::State& state, uint y, uint numSpace);
+    void prepareDailyOptimalGenerations(Solver::Variable::State& state, uint y, uint numSpace, VAL_GEN_PAR_PAYS& valeursGenereesParPays);
```

Comment:
We should pass only the data which correspond to this numspace

Reply:
We could still go in the right direction by passing only the necessary data, although numspace is needed for other data.

But ok, let's tackle it later with a more targeted evolution about spaces (https://github.com/AntaresSimulatorTeam/Antares_Simulator/issues/1571).